### PR TITLE
Fix for Mac: replace all instances of modal dialog Hide() with EndModal(0).

### PR DESCRIPTION
**Pull request by ioan-chera**
They would never return out of ShowModal, even if the dialogs were closed/hidden, so the program would become unresponsive.

__Unless these changes are obviously correct, please test this before merging, because I didn't have time to verify it on Windows__

### CODE REVIEW

**Comment by GitExl** [comment]
Thanks! Does this mean WhackEd4 works on OSX now?

**Comment by ioan-chera** [comment]
Not yet, because there are some strange issues with the MDI child frames only showing up when the user clicks on the dock icon. OSX has no such things as MDI child windows (dunno about Linux).

Also when clicking the main window close button, only the window closes, and I'm unable to close the app other than by force-quitting.

My second commit added several "None" and bounds checks in the code, just to prevent the errors from exiting the program. I think they'll have to be replaced with something more fitting if the OS is Mac. What I basically did there was bypass invalid states (for OSX).

The UI aspect doesn't seem as polished, for example I can't see the toolbar buttons shading nicely. But that's not so important. What matters is a working GUI. I can save files at least, but I have the aforementioned glitches.

I used py2app to generate the "app" package so I don't have to run Python manually.

So in short, it's not ready yet. Thanks for upgrading to wxPython to 3.0. The previous dependency didn't work on Mac, it would crash all the time.

**Comment by GitExl** [comment]
I was aware of the MDI issue, but I had hoped that would be the only real issue. Maybe the app not closing after exiting the main Window can be fixed by commenting out src/whacked4/app.py:51? Maybe https://wiki.wxwidgets.org/WxMac-specific_topics#When_to_close_the_program is helpful too.

At any rate, I'll gladly accept any future pull requests for OSX support. Maybe in time a proper OSX binary release will be possible.
